testplan/common/utils/monitor/collectors.py

```diff
@@ -0,0 +1,121 @@
+        self._io_counter = {}
+
+    def monitor(self):
+        parent = psutil.Process(self._pid)
+        procs = parent.children(recursive=True)
+        procs.append(parent)
+        monitor_result = {
+            'cpu': 0,
+            'memory': 0,
+            'disk': psutil.disk_usage(self.metadata['disk_director']).percent,
+            'iops': 0,
+            'read': 0,
+            'write': 0,
+        }
+        for proc in procs:
+            try:
+                raw_data = proc.as_dict(
+                    attrs=['pid', 'name', 'memory_info', 'cpu_percent', 'io_counters', 'create_time', 'io_counters'])
+
+            except psutil.NoSuchProcess:
+                continue
+            monitor_result['cpu'] += raw_data['cpu_percent'] if raw_data['cpu_percent'] > 0 else 0
+            monitor_result['memory'] += raw_data['memory_info'].rss
+            try:
+                counters = {
+                    'io_counter': raw_data['io_counters'].read_count + raw_data['io_counters'].write_count,
+                    'read_counter': raw_data['io_counters'].read_bytes,
+                    'write_counter': raw_data['io_counters'].write_bytes,
+                    'time': time.time()
+                }
+                iops = read = write = 0
+                if raw_data['pid'] in self._io_counter \
+                        and self._io_counter[raw_data['pid']]['io_counter'] >= counters['io_counter']:
+                    _interval = counters['time'] - self._io_counter[raw_data['pid']]['time']
+                    iops = (counters['io_counter'] - self._io_counter[raw_data['pid']]['io_counter']) / _interval
+                    read = (counters['read_counter'] - self._io_counter[raw_data['pid']]['read_counter']) / _interval
+                    write = (counters['write_counter'] - self._io_counter[raw_data['pid']]['write_counter']) / _interval
+                else:
+                    _interval = counters['time'] - raw_data['create_time']
+                    if _interval >= 1.0:
+                        iops = counters['io_counter'] / _interval
+                        read = counters['read_counter'] / _interval
+                        write = counters['write_counter'] / _interval
+                self._io_counter[raw_data['pid']] = counters.copy()
+                monitor_result['iops'] += iops
+                monitor_result['read'] += read
+                monitor_result['write'] += write
+            except (AttributeError, KeyError):
+                # process exited or no io counters
+                pass
+        return monitor_result
+
+
+class LinuxCollector(Collector):
+    def __init__(self, director):
+        super(LinuxCollector, self).__init__(director)
+        try:
+            cpu_info = subprocess.check_output(['lscpu'])
+            self.metadata['extra']['cpu_info'] = cpu_info.decode()
+        except Exception:
+            pass
+
+
+class LinuxContainerCollector(LinuxCollector):
```

Comment:
Is this relevant to OSS world? Are the file paths standard? should they be configurable?